templates/subscription-centre/index.html

```diff
@@ -86,45 +104,169 @@ <h2 class="p-modal__title" id="modal-title">Unsubscribe</h2>
+      returnUrl.value = "#unsubscribe";
+      form.submit();
+    };
+
+    const enableUpdatePreferencesBtn = () => {
+      const updatePreferencesBtn = document.querySelector("#update-preferences");
+      updatePreferencesBtn.disabled = false;
+    };
+
+    document.querySelector("#update-preferences").addEventListener("click", handleUpdatePreferencesBtn, { once: true });
+
+    document.querySelector("#unsubscribe-btn").addEventListener("click", handleUnsubscribeBtn, { once: true });
 
     form.addEventListener('input', () => {
       if (hasRun) return;
       hasRun = true;
       enableUpdatePreferencesBtn();
     });
-  });
+  })()
+</script>
+<script>
+  // Setup search
+  (function() {
+    const subsContainer = document.querySelector("#subscriptions-list");
+    const subsSearch = document.querySelector("#subscription-centre-search");
+    const resultContainer = document.querySelector(".p-search-and-filter__panel");
+    const searchInput = document.querySelector("#subscription-search");
+
+    const attachListeners = () => {
+      const clearSearchBtn = subsSearch.querySelector(".p-search-and-filter__clear");
+
+      searchInput.addEventListener("input", handleSearch);
 
+      clearSearchBtn.addEventListener("pointerdown", handleClearBtn);
+      clearSearchBtn.addEventListener("keyup", handleClearBtn);
+
+      resultContainer.addEventListener("pointerdown", handleSearchSelection);
+      resultContainer.addEventListener("keyup", handleSearchSelection);
+
+      //subsSearch.addEventListener("focusout", hideResultsDropdown);
+      searchInput.addEventListener("focus", (e) => e.target.value && showDropdown());
+    };
+
+    // Get all input field and map them into a searchable array
+    const buildSearchableList = () => {
+      const nodeList = subsContainer.querySelectorAll("input[type=checkbox]");
+      const subsArray = [].map.call(nodeList, node => node.value);
+
+      return subsArray;
+    };
+    
+    // Searches for matching strings and renders the matches in the search dropdown
+    const handleSearch = (e) => {
+      let searchVal = e.target.value.trim().toLowerCase();
+
+      const renderResults = (results) => {
+        const templateContainer = document.querySelector("#search-result-template");
+
+        clearResults();
+        showDropdown();
+
+        if (!results.length) {
+          results.push(`No results for ${searchVal}...`);
+        } else {
+          results.slice(0, 10).forEach(result => {
+            const newResultNode = templateContainer.content.cloneNode(true);
+            newResultNode.querySelector("span").innerText = result;
+            resultContainer.appendChild(newResultNode);
+          });
+        }
+      };
+
+      // Checks for exact match of string and add to the top of results. If no exact match, look for any kind of match and add to the end of results
+      if (searchVal) {
+        const searchResults = [];
+        for (let i = 0; i < subsList.length; i++) {
```

Comment:
As you're using ES6 syntax, and we don't need to worry about supporting IE, you could use `.forEach` here if you wanted, rather than a for loop.